neonvm/controllers/catch_panic.go

```diff
@@ -0,0 +1,33 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"runtime/debug"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type catchPanicReconciler struct {
+	inner reconcile.Reconciler
+}
+
+func withCatchPanic(r reconcile.Reconciler) reconcile.Reconciler {
+	return &catchPanicReconciler{inner: r}
+}
+
+func (r *catchPanicReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+	log := log.FromContext(ctx) // TODO: is this valid?
```

Comment:
Not sure what is the answer to this question, but `log.FromContext` returns a dummy logger that does nothing if ctx is nil or it doesn't have a logger. Since Reconcile is being called by the controller manager its context *should have* a valid logger.